### PR TITLE
Align optimizer

### PR DESCRIPTION
**Pull request by gcroci2**
In NeuralNet class, the optimizer used during training is Adam and can't be set by the user.

With this PR the user can set the optimizer using `configure_optimizer` method of NeuralNet class. 
The default is Adam, and I changed the default learning rate from 0.01 to 0.001 (the default one, see [PyTorch documentation](https://pytorch.org/docs/stable/generated/torch.optim.Adam.html)).

Refers to issue #154

### CODE REVIEW

**Review by cbaakman — APPROVED** [review]
It seems OK to have the optimizer configuration in a different class method.
